src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import "./Modal.css";
+import ImageUploading, { ImageListType } from "react-images-uploading";
+import { Post } from "../App/App";
+// import { idText } from "typescript";
+
+interface ModalProps {
+  submitPost: (newPost: Post) => void;
+  closeModal: (event: any) => void;
+}
+
+export const Modal: React.FC<ModalProps> = ({ submitPost, closeModal }) => {
+  const [images, setImages] = useState([]);
+  const [postTitle, setTitle] = useState<string>("");
+  const [postDescription, setDescription] = useState<string>("");
+  const [imageURLS, setImageURLS] = useState<string[]>([]);
+
+  const maxNumber = 2;
+  const onChange = (
+    imageList: ImageListType,
+    addUpdateIndex: number[] | undefined
+  ) => {
+    setImages(imageList as never[]);
+    let onlyURLs = imageList.map((image) => image.data_url);
+    setImageURLS(onlyURLs);
+  };
+
+  const clearState = () => {
+    setImages([]);
+    setTitle("");
+    setDescription("");
+  };
+
+  const addPost = (event: React.MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    const newPost = {
+      id: 2,
```

Comment:
Is this just "2" because you're using test data? Or should that be a variable?

Reply:
It will be created by you eventually, we just needed to get it working for the moment

Reply:
This is just dummy data - claire and I were going to cross this bridge when we actually connect FE to BE